packages/core/src/components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,214 @@
+  }
+
+  render() {
+    const body = document.querySelector('body');
+    return (
+      <Manager>
+        <Reference>
+          {({ ref }) => (
+            <button
+              type="button"
+              onTouchStart={() => this.showTooltip()}
+              onFocus={() => this.handleFocus()}
+              onBlur={
+                this.props.hasInteractiveContent
+                  ? null
+                  : () => this.hideTooltip()
+              }
+              onMouseEnter={() => this.showTooltip()}
+              onMouseLeave={() => this.hideTooltip()}
+              id={this.props.id}
+              aria-label={`Tooltip: ${this.props.ariaLabel || ''}`}
+              className={classNames(
+                'ds-c-tooltip__trigger',
+                this.props.triggerClasses
+              )}
+              ref={ref}
+            >
+              {this.props.triggerContent}
+              <TooltipIcon
+                hasTriggerContent={this.props.triggerContent != null}
+                iconClasses={this.props.iconClasses}
+                inverse={this.props.inverse}
+                showTooltip={this.state.showTooltip}
+              />
+            </button>
+          )}
+        </Reference>
+        {body !== null &&
+          // Need to use portal or else positioning is off in IE11 sometimes
+          ReactDOM.createPortal(
+            <Transition
+              in={this.state.showTooltip}
+              unmountOnExit
+              timeout={transitionDuration}
+            >
+              {transitionState => (
+                <Popper
+                  positionFixed={this.props.positionFixed}
+                  placement={this.props.placement}
+                  modifiers={{ offset: { offset: TOOLTIP_OFFSET_OPT } }}
+                >
+                  {({ placement, ref, style, arrowProps }) => {
+                    // Need to add back 1/2 width of arrow to the left placement of the
+                    // tooltip container so the arrow shows up at exactly 50% as the
+                    // arrow container has a width/height we are setting that the
+                    // arrowProps positioning here does not account for
+                    let leftArrowOffset = 0;
+                    if (parseInt(arrowProps.style.left, 10)) {
+                      leftArrowOffset = arrowProps.style.left + 8;
+                    }
+                    const arrowStyle = { left: leftArrowOffset };
+
+                    // Can't directly modify style, so copy and add styles from props
+                    const newStyle = {
+                      ...style,
+                      ...defaultTransitionStyle,
+                      ...transitionStyles[transitionState]
+                    };
+                    newStyle.maxWidth = this.props.tooltipMaxWidth || '300px';
+                    newStyle.zIndex = this.props.tooltipZIndex || '1';
+                    return (
+                      <div
```

Comment:
We may need to add a `role=tooltip` attribute here for the `hasInteractiveContent=false` case.  For Tooltip dialogs with interactive content, I think the recommendation is to use `role=dialog`.